### PR TITLE
serverless/appsec: upgrade to go-libddwaf v2.2.0

### PR DESCRIPTION
**Pull request by Hellzy**
### What does this PR do?

- Upgrade to v2.2.0 of go-libddwaf
- Upgrade to v1.0.2 of appsec-internal-go
- Adjust integration tests payloads to trigger the expected SQL injection rule

**API Changes**:
- WAF now returns events as as an array of go `any` values instead of `json.RawMessage`.

- WAF run return values are now aggregated in a new WAF `Result` struct

### Motivation

Newer WAF functionalities are needed for feature development (request schemas extraction for API security)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/appsec/httpsec/http.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ import (
 // subprocessor monitoring the given security rules addresses and returning
 // the security events that matched.
 type Monitorer interface {
-	Monitor(addresses map[string]interface{}) (events []byte)
+	Monitor(addresses map[string]interface{}) (events []any)
```
Comment (RomainMuller) [inline pkg/serverless/appsec/httpsec/http.go:33]:
```suggestion
	Monitor(addresses map[string]any) (events []any)
```

File: pkg/serverless/appsec/appsec.go [hunk 2]
```diff
@@ -122,15 +121,16 @@ func (a *AppSec) Monitor(addresses map[string]interface{}) (events []byte) {
 			return nil
 		}
 	}
+
 	dt, _ := ctx.TotalRuntime()
-	if len(events) > 0 {
-		log.Debugf("appsec: security events found in %s: %s", time.Duration(dt), string(events))
+	if len(res.Events) > 0 {
```
[the remark this answers is not in the document]
Comment (Hellzy) [reply pkg/serverless/appsec/appsec.go:126]: If it's a method it should be done in go-libddwaf.
See DataDog/go-libddwaf#41

File: pkg/serverless/appsec/httpsec/http.go (hunk 1 above)
Comment (Hellzy) [reply pkg/serverless/appsec/httpsec/http.go:33]: [10a1323](https://github.com/DataDog/datadog-agent/pull/20540/commits/10a13239cf656081c51280c0fea98edd63209bb4)

File: pkg/serverless/appsec/appsec.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hellzy) [reply pkg/serverless/appsec/appsec.go]: b9ddca4dbcbdfa45

**Comment by Hellzy** [comment]
b9ddca4dbcbdfa45 Currently upgrades to 1.15.0 with a dangling go-libddwaf version.
This will be updated when we tag for go-libddwaf v2.0.0

File: pkg/serverless/appsec/appsec.go [hunk 3]
```diff
@@ -103,17 +103,16 @@ func (a *AppSec) Close() error {
 	return nil
 }
 
-// Monitor runs the security event rules and return the events as raw JSON byte
-// array.
-func (a *AppSec) Monitor(addresses map[string]interface{}) (events []byte) {
+// Monitor runs the security event rules and return the events as an slice
+func (a *AppSec) Monitor(addresses map[string]any) []any {
```
Comment (RomainMuller) [inline pkg/serverless/appsec/appsec.go:107]: This should be replaced with a version that allows for ephemeral and permanent addresses to be provided (possibly accepting a `waf.RunAddressData` directly?)

File: pkg/serverless/appsec/httpsec/http.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ import (
 // subprocessor monitoring the given security rules addresses and returning
 // the security events that matched.
 type Monitorer interface {
-	Monitor(addresses map[string]interface{}) (events []byte)
+	Monitor(addresses map[string]any) (events []any)
```
Comment (RomainMuller) [inline pkg/serverless/appsec/httpsec/http.go:33]: This should allow permanent & ephemeral.

**Review by purple4reina — APPROVED** [review]

**Review by eliottness — APPROVED** [review]
LGTM :+1:

**Review by kkhor-datadog — APPROVED** [review]

**Pull request merged**